share/functions/fish_md5.fish

```diff
@@ -1,7 +1,7 @@
 function fish_md5
     if type -q md5sum
         # GNU systems
-        echo (md5sum $argv[1] | string split ' ')[1]
+        echo (echo $argv[1] | md5sum | string split ' ')[1]
```

Comment:
Sorry, but this would break the other use (which should also ideally use this function), and it's not consistent with what `md5` below does.

If you don't pass an argument, this will end up reading the terminal (which I'm not sure it should), so you'll need to either

- Introduce an option that it should md5sum the given string, instead of treating it as a filename

- Read from stdin if no argument is given

- Use `psub` (not recommended)